File: JULY 23/31 July/BFS of graph gfg.cpp
class Solution {
  public:
    // Function to return Breadth First Traversal of given graph.
    vector<int> bfsOfGraph(int V, vector<int> adj[]) {
        // Code here
        int k=0;
        vector<int> bfsTraversal, visited(V);
        
        queue<int> q;
        q.push(0);
        visited[0] = 1;
        while(!q.empty())
        {
            int node = q.front();
            bfsTraversal.push_back(node);
            q.pop();
            for(int child:adj[node])
            {
                if(visited[child]==0)
                {
                    visited[child] = 1;
                    q.push(child);
                }
            }
        }
        
        return bfsTraversal;
    }
};

File: APRIL 2025/02 April/BFS of graph gfg.java
// User function Template for Java
class Solution {
    // Function to return Breadth First Search Traversal of given graph.
    public ArrayList<Integer> bfs(ArrayList<ArrayList<Integer>> adj) {
        // code here
        ArrayList<Integer> traversal = new ArrayList<>();
        Queue<Integer> q = new ArrayDeque<>();
        boolean[] visited = new boolean[adj.size()];
        q.offer(0);
        visited[0] = true;
        while(!q.isEmpty()) {
            int node = q.poll();
            traversal.add(node);
            for(Integer next:adj.get(node)) {
                if(visited[next])
                    continue;
                q.offer(next);
                visited[next] = true;
            }
        }
        
        return traversal;
        
    }
}
